src/main/java9/org/jboss/modules/ModuleLoggerFinder.java

```diff
@@ -88,7 +89,7 @@ public System.Logger getLogger(final String name, final java.lang.Module module)
      *
      * @param cl the class loader to use
      */
-    static void activate(final ClassLoader cl) {
+    public static void activate(final ClassLoader cl) {
```

Comment:
I'd rather not expose this API; what is the use case?

Reply:
There are cases when, for example the WildFly Bootable JAR, we don't use `org.jboss.modules.Main` as the entry point so `activate()` is never invoked. It would be the responsibility of those callers which do not launch through the main entry point to invoke this. I do realize that is a bit odd. I'm definitely open to other options.

https://github.com/wildfly/wildfly-core/pull/5882 is what I'm doing in WildFly Core for now.

Reply:
Reflection is probably still best for now. What we need to do is to create an embedding API in JBoss Modules and have a little bit more of a control over the bootstrapping process. Right now most important setup tasks are hard-coded in the `main` method. So ideally, we'd have a bootstrap API with error checking (e.g. "hey you already bootstrapped this once") and then the `main` method would reuse it. Maybe we could do some kind of design session on this in the near future.

Reply:
That makes a lot of sense to me. I'll pull the part of making the API public out, but I do think having the queue length somewhat auto-enable it makes some sense. What do you think of that part @dmlloyd?

Reply:
Having a maximum queue length does make sense (we've done similar things in other contexts to avoid queue runaway). But I don't know if activation is the right thing to do in response to hitting the maximum. It seems more like we're just hoping that some large-ish queue depth represents that we've waited "long enough" for things to go through, doesn't it?

Reply:
Yes, the number is completely arbitrary for sure. I guess it would likely be strange to dump a bunch of logs. The idea of activating it was just to have the `System.Logger`'s work rather than loosing logs.

Reply:
The introduction of JBoss Modules bootstrap API is a brilliant idea to solve kind of issues @jamezp is facing @dmlloyd ! When you'll schedule the design meeting please keep me in the loop ;)

Reply:
I just discovered there's a JIRA for it already: https://issues.redhat.com/browse/MODULES-367

Reply:
Goof find @ropalka. I've linked the two JIRA's together.